test/utils.ts

```diff
@@ -66,3 +69,106 @@ export async function ensureEditorServicesIsConnected(): Promise<IPowerShellExte
+
+        // BUG: resolveCliArgs misdetects the extensions/userdata folder sometimes, we need to fix that.
+        // regex that matches .vscode-test\\vscode*\\.vscode-test in an OS independent way
+        const vscodeIncorrectPathRegex = /\.vscode-test[\\/]vscode.+?[\\/]\.vscode-test/;
+        args[0] = args[0].replace(vscodeIncorrectPathRegex, ".vscode-test");
+        args[1] = args[1].replace(vscodeIncorrectPathRegex, ".vscode-test");
+
+        //TODO: This is the best way I could come up with to wait for the C# extension to activate, I'm sure there's a more terse way to do this.
+        let cSharpActivatedResolved: (value: string) => void;
+        const waitForCSharpExtensionActivation = new Promise<string>((resolve) => {
+            cSharpActivatedResolved = resolve;
+        });
+        const testCSharpActive = async (): Promise<void> => {
+            const csharpExtension = vscode.extensions.getExtension("ms-dotnettools.csharp");
+            if (csharpExtension) {
+                cSharpMonitorEvent.dispose();
+                await csharpExtension.activate();
+                console.log("C# Extension is now active");
+                cSharpActivatedResolved("ACTIVATED");
+            }
+        };
+        const cSharpMonitorEvent = vscode.extensions.onDidChange(testCSharpActive);
+
+        // HACK: There is no API for this so we are forced to use the code CLI method. This is actually what is recommended from the resolveCliArgsFromVSCodeExecutablePath docs.
+        const installExtensionArgs = [...args, "--install-extension", "ms-dotnettools.csharp"];
+        console.log(`Starting Extension Install: ${cli} ${installExtensionArgs.join(" ")}`);
+
+        const { status, stdout, stderr} = spawnSync(cli, installExtensionArgs, {
+            encoding: "utf-8",
+            windowsHide: true
+        });
+        if (status !== 0 || !stdout.match(/csharp.+was successfully installed/)) {
+            throw new Error(`Failed to install C# extension: ${stdout} ${stderr}`);
+        }
+        console.log(stdout);
+
+        console.log("Waiting for C# extension activation");
+        // Wait for the csharp extnesion to be activated
+        if (await waitForCSharpExtensionActivation !== "ACTIVATED") {
+            throw new Error("Failed to activate C# extension");
+        }
+        console.log("==INSTALLED: C# Extension==");
+    }
+}
+
+/** Builds the sample binary module code. We need to do this because the source maps have absolute paths so they are not portable between machines, and while we could do deterministic with source maps, that's way more complicated and everywhere we build has dotnet already anyways */
+export function BuildBinaryModuleMock(): void {
+    console.log("==BUILDING: Binary Module Mock==");
+    const projectPath = path.resolve(`${__dirname}/../../test/mocks/BinaryModule/BinaryModule.csproj`); //Relative to "out/test" when testing.
+    const buildResult = execSync(`dotnet publish ${projectPath}`);
```

Comment:
The CI pipeline _won't_ have `dotnet` when it's doing a release (because it's building from downloaded PSES artifacts instead of from source). We'll need to skip these tests in that case.

Reply:
https://github.com/PowerShell/vscode-powershell/blob/35fcd161b04d0e94c873a631b3d3502e4c93de84/.vsts-ci/templates/ci-general.yml#L43-L44

Reply:
@andschwa it's still going to have dotnet from the pipeline image. We don't need latest and greatest because we're building netstandard.

Reply:
Ok sweet.